### PR TITLE
Update Banklink code for new banklink specification. +LHV Hire functionality.

### PR DESCRIPTION
**Pull request by ttambet**
- Update Banklink code for new banklink specification.
- Implement Hire LHV functionality.

### CODE REVIEW

File: wirexfers/providers/hire.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+"""
+    wirexfers.protocols.hire
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (plaes) [inline wirexfers/providers/hire.py]: underline length..

File: wirexfers/providers/hire.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    LHV hire banklink implementations.
+
+    :copyright: (c) 2015, Timo Tambet
```
Comment (plaes) [inline wirexfers/providers/hire.py]: 2016 :)

File: wirexfers/providers/hire.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+from Crypto import Random
+from Crypto.Hash import SHA
+from Crypto.Signature import PKCS1_v1_5
+
+from . import KeyChainBase, ProviderBase
+from .. import PaymentResponse
+from ..exc import InvalidResponseError
+
+class EELHVProvider(ProviderBase):
+    """LHV IPizza hire panklink provider.
```
Comment (plaes) [inline wirexfers/providers/hire.py]: bank :)

File: wirexfers/providers/hire.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            raise InvalidResponseError
+
+        success = resp in fields
+        Random.atfork()
+
+        # Parse and validate date
+        f = lambda x: form.get('VK_%s' % x)
+        t = datetime.strptime(f('DATETIME').split('+')[0], "%Y-%m-%dT%H:%M:%S")
+
+        if datetime.now() - timedelta(seconds=300) > t:
```
Comment (plaes) [inline wirexfers/providers/hire.py:88]: Would be nice to have it documented somewhere (that timeout is 300 seconds..)

Maybe even use TimeoutError exception, so one can show proper error message in application.

File: wirexfers/providers/ipizza.py [hunk 5]
```diff
@@ -96,19 +102,25 @@ def parse_response(self, form, success=True):
         if not PKCS1_v1_5.new(self.keychain.public_key) \
                          .verify(SHA.new(m), b64decode(f('MAC'))):
             raise InvalidResponseError
+
+        timestamp = datetime.strptime(f('T_DATETIME'), '%Y-%m-%dT%H:%M:%S+0300')
+        now = datetime.now()
+        if timestamp > (now + timedelta(seconds=300)) or timestamp < (now - timedelta(seconds=300)):
+            raise Exception('Response outdated.')
```
Comment (plaes) [inline wirexfers/providers/ipizza.py]: See also previous comment about using TimeoutError

File: wirexfers/providers/ipizza.py [hunk 6]
```diff
@@ -96,19 +102,25 @@ def parse_response(self, form, success=True):
         if not PKCS1_v1_5.new(self.keychain.public_key) \
                          .verify(SHA.new(m), b64decode(f('MAC'))):
             raise InvalidResponseError
+
+        timestamp = datetime.strptime(f('T_DATETIME'), '%Y-%m-%dT%H:%M:%S+0300')
+        now = datetime.now()
+        if timestamp > (now + timedelta(seconds=300)) or timestamp < (now - timedelta(seconds=300)):
```
Comment (plaes) [inline wirexfers/providers/ipizza.py]:
```
if not ((now + timedelta(seconds=300)) < timestamp < (now - timedelta(seconds=300))):
   raise TimeoutError
```

File: wirexfers/providers/ipizza.py (hunk 5 above)
Comment (van2048) [reply wirexfers/providers/ipizza.py]: Sample data returned in the server response: 
u'VK_T_DATETIME': [u'2016-10-24T00:14:17+0300']
Sample host data in this moment:
now = datetime.datetime(2016, 10, 23, 21, 14, 17, 535072)
Probably a problem with the time zones.

Comment (plaes) [reply wirexfers/providers/ipizza.py]: `datetime.now()` returns local time without timezone information.

We probably have to add `pytz` as dependency and use `datetime.now(pytz.utc)` instead.

**Comment by plaes** [comment]
Can you create v2 of this patchset as a new pull request where you only include banklink stuff and rebase-squash commits?

File: wirexfers/providers/ipizza.py [hunk 7]
```diff
@@ -103,10 +107,17 @@ def parse_response(self, form, success=True):
...
+        d = parse(f('T_DATETIME')).astimezone(pytz.utc)
+
+        # Current time (UTC)
+        now = datetime.now(pytz.utc)
+
+        # Timedelta 5 min
+        td = timedelta(seconds=300)
+        if not ((now - td) < d < (now + td)):
+            # FIXME: Python 3 should support Timeout exception
+            raise requests.exceptions.Timeout
```
Comment (plaes) [inline wirexfers/providers/ipizza.py:120]: Why requests? :)

I would create new class inside wirexfers for that, like this:

https://github.com/plaes/python-erply-api/blob/master/erply_api.py#L30

```
class WirexfersTimeoutException(Timeout):
    pass
```

**Pull request closed without merging**